substrate-sgx/externalities/src/lib.rs

```diff
@@ -15,32 +15,39 @@
 */
 #![cfg_attr(not(feature = "std"), no_std)]
 
-#[cfg(not(feature = "std"))]
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
```

Comment:
Is it really necessary to always specify `not(feature = "std"), feature = "sgx"` when we've already ensured above that if feature == sgx, then feature std must not be set?

i.e. would it no be enough to specify `#[cfg(feature = "sgx")]`?

Reply:
I've had problems with this in the past, when I didn't explicitly exclude one and include the other.. I don't know why, I would also think that it's enough without the `not(feature =`

Reply:
I think you have more experience with this matter than I do. So I'm leaving this decision to you.

Reply:
I will try anyway 👍

Reply:
Seems to work fine, so thanks for nagging 😄